packages/test/test-version-utils/package.json

```diff
@@ -29,7 +29,7 @@
 		"prettier": "prettier --check . --ignore-path ../../../.prettierignore",
 		"prettier:fix": "prettier --write . --ignore-path ../../../.prettierignore",
 		"test": "npm run test:mocha",
-		"test:mocha": "mocha  --recursive \"dist/test/*.spec.js\" --exit --project src/test/tsconfig.json -r node_modules/@fluidframework/mocha-test-setup",
+		"test:mocha": "mocha  --recursive \"dist/test/*.spec.js\" --exit --project src/test/tsconfig.json -r node_modules/@fluidframework/mocha-test-setup --timeout=10000",
```

Comment:
Unfortunately, these tests can take a few seconds since we now pull in version data from npm. Not sure what workarounds we can do other than pre-fetching all (or most) existing versions at the start somehow

Reply:
Is it fetching version data only, or is it also pulling down packages?

Reply:
Only fetching version data, and I don't think this will have a large impact on our actual e2e tests. But specifically for the `test-version-utils` unit tests we make a lot of these calls back-to-back-to-back, which is why I extended the timeout

Reply:
Can you point me to the code that calls npm? Seems like the version-tools getPreviousVersions or similar functions could speed this up by querying git for the versions instead of npm.

Reply:
Sure, you can find the call here:
https://github.com/microsoft/FluidFramework/blob/c2aa13a1d1989512e8d3eb407b3212dd47509179/packages/test/test-version-utils/src/versionUtils.ts#L138C7-L138C7

Reply:
Thanks. This package might be faster (https://www.npmjs.com/package/latest-version) but maybe not. The name is misleading - you can pass it ranges and it returns the latest matching version. There's also https://www.npmjs.com/package/package-json. Anyway might be worth testing to see if it's any better. It's at least less code.

Reply:
I think it could be nice, but also requires us to make `getRequestedVersion()` asynchronous, which turns into a bit of a headache since everything that calls it is also not async 😅

Reply:
Like I said, I don't think this will really affect our e2e test suite run time, but if we do want to improve it for the the unit tests I think the best approach could be something like what you suggested in another comment:

>There's a function in version-tools, getPreviousVersions, that might be worth looking at to see if there's any overlap.
Also, this may be a little far-fetched, but I am wondering if the logic for this gets easier if you had a sorted list of all the versions we released? If you had an array with all the versions, then you could find the base version in the list, then take array[baseVersionIndex - 2], for example. If it would simplify things, such a list can be built using getPreviousVersions.

I'm not sure if it's worth it to go after this approach right now given our time constraints but I don't see any glaring issues with this approach and I think it could simplify things a lot